met-api/src/met_api/resources/staff_user.py

```diff
@@ -176,3 +182,35 @@ def get(user_id):
             return jsonify(engagement_schema.dump(members, many=True)), HTTPStatus.OK
         except BusinessException as err:
             return {'message': err.error}, err.status_code
+
+@cors_preflight('GET,OPTIONS')
+@API.route('/<user_id>/tenants')
+class UserTenants(Resource):
+    """Fetches tenants for a given user."""
+
+    @staticmethod
+    @cross_origin(origins=allowedorigins())
+    @auth.requires_auth
+    def get(user_id):
+        """Get tenant details by user id."""
+        if(user_id == 'me'):
+            user_data = TokenInfo.get_user_data()
+            user_id = user_data.get('external_id')
+            print("User ID: ", user_id)
+            user_roles = current_app.config['JWT_ROLE_CALLBACK'](g.jwt_oidc_token_info)
+            if Role.SUPER_ADMIN.value in user_roles:
+                return TenantService.get_all(), HTTPStatus.OK
+        else:
+            authorization.check_auth(
```

Comment:
I guess if you're a super admin, you will get a full dropdown of possible tenants, and if you are just a viewer, you can see your own tenant? Makes sense.